docs/requirements.txt

```
@@ -1,6 +1,6 @@
 absl-py
 ipython>=8.8.0  # 8.7.0 has ipython3 lexer error
-sphinx>=7.3.2  # 7.3.0 breaks sphinx-book-theme
+sphinx~=7.4.0  # 7.3.0 breaks sphinx-book-theme
```

Let's make this more specific:
```suggestion
sphinx>=7.3.2,<8.0 # 7.3.0 breaks sphinx-book-theme; 8+ triggers warnings in myst_nb v1.1
```

---

I'm making this change separately in #22896 in order to unbreak doc builds at head.

---

Thanks @jakevdp! Rebased, testing...